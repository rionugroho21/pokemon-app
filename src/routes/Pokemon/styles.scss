.pokemon {
  position: relative;
  width: 100%;
  max-width: 1020px;
  margin: 0px auto;
  display: block;
  padding: 10px 16px 30px 16px;

  &-nav {
    overflow: auto;
    margin-bottom: 20px;

    &-total {
      font-size: 14px;
      margin-top: 6px;
      float: left;
    }

    &-pagination {
      text-align: center;
      float: right;

      button {
        padding: 5px 18px;
        border: none;
        background-color: #a9a9a9;
        border-radius: 3px;
        cursor: pointer;
        color: #000;
        font-weight: 400;
        font-size: 14px;

        &:last-child {
          margin-left: 10px;
        }

        &:hover {
          background-color: #cccccc;
        }
      }
    }
  }

  &-list{
    &-item {
      position: relative;
      width: 22%;
      font-size: 14px;
      padding: 0px;
      border: 1px solid #dedede;
      border-radius: 4px;
      margin: 0px 4% 25px 0px;
      display: inline-block;
      text-align: center;
      cursor: pointer;
      text-transform: capitalize;
      text-decoration: none;
      color: #000;
      opacity: 0.95;
      box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
      transition: all 0.3s cubic-bezier(0.25, 0.8, 0.25, 1);
      &:hover {
        box-shadow: 0 14px 28px rgba(0, 0, 0, 0.25), 0 10px 10px rgba(0, 0, 0, 0.22);
        text-decoration: none;
      }
      -moz-user-select: none;
      -website-user-select: none;
      -ms-user-select: none;
      user-select: none;
      -o-user-select: none;

      &:nth-child(4n) {
        margin-right: 0px;
      }

      p {
        position: absolute;
        top: 5px;
        right: 5px;
        font-size: 12px;
        background-color: #a7daff;
        padding: 4px 5px;
        border-radius: 3px;
      }

      &-image {
        width: 100%;
        height: auto;
        min-height: 30px;
        padding: 20px 0px;
      }

      img {
        width: 110px;
        height: auto;
        margin: 0px auto;
      }

      &-title {
        width: 100%;
        display: block;
        background-color: #e8e8e8;
        padding: 10px 0px;
      }
    }
  }
}

@media only screen and (max-width: 1024px) {
  .pokemon {
    width: 100%;
    max-width: 100%;

    &-list{
      &-item {
        width: 48%;
        margin: 0px 4% 15px 0px;

        &:nth-child(4n) {
          margin-right: 4%;
        }

        &:nth-child(2n) {
          margin-right: 0;
        }
      }
    }
  }
}

@media only screen and (max-width: 640px) {
  .pokemon {
    &-list{
      &-item {
        width: 100%;
      }
    }
  }
}