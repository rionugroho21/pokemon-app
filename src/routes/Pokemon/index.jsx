import React from 'react';
import { Link } from 'react-router-dom';

import get from 'lodash/get';

import usePokemon from './hooks/usePokemon';

import pokeball from '../../assets/images/pokeball.png';

import './styles.scss';

export const Pokemon = props => {
  const {
    loading,
    listPokemon,
    nextPageUrl,
    prevPageUrl,
    gotoNextPage,
    gotoPrevPage,
    handleCheckOwned
  } = usePokemon(props);

  return (
    <>
      {loading ? (
        <p className="loading">Loading...</p>
      ) : (
        <div className="pokemon">
          <div className="pokemon-nav">
            <p className="pokemon-nav-total">Total {listPokemon.length}</p>
            <div className="pokemon-nav-pagination">
              {prevPageUrl && <button onClick={() => gotoPrevPage(prevPageUrl)}>Previous</button>}
              {nextPageUrl && <button onClick={() => gotoNextPage(nextPageUrl)}>Next</button>}
            </div>
          </div>

          <div className="pokemon-list">
            {listPokemon.map((list, key) => {
              // const url = get(list, 'url', '');
              // const pokemonIndex = url.split('/')[url.split('/').length - 2];
              // const imageUrl = `https://github.com/PokeAPI/sprites/blob/master/sprites/pokemon/${pokemonIndex}.png?raw=true`;
              const owned = handleCheckOwned(list.name);

              return (
                <Link to={`/pokemon/${list.name}`} className="pokemon-list-item" key={key}>
                  <p>Owned : {owned}</p>
                  <div className="pokemon-list-item-image">
                    <img src={pokeball} alt="Pokeball" width="110" />
                  </div>
                  <div className="pokemon-list-item-title">
                    {list.name}
                  </div>
                </Link>
              );
            })}
          </div>
        </div>
      )}
    </>
  );
};

export default Pokemon;