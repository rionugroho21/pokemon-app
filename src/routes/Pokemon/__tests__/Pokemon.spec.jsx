import React from 'react';
import { render, cleanup } from '@testing-library/react';

import Pokemon from '../index';

const props = {
  loading: false,
  listPokemon: [],
  nextPageUrl: jest.fn(),
  prevPageUrl: jest.fn(),
  gotoNextPage: jest.fn(),
  gotoPrevPage: jest.fn(),
  handleCheckOwned: jest.fn()
};

describe('Pokemon Page', () => {
  afterEach(cleanup);

  it('should render page correctly', () => {
    const { asFragment } = render(<Pokemon {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});