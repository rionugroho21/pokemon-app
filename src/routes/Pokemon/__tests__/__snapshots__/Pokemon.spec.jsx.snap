// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon Page should render page correctly 1`] = `
<DocumentFragment>
  <p
    class="loading"
  >
    Loading...
  </p>
</DocumentFragment>
`;
